File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"
#include <cmath>

namespace darts {
    int score(double x, double y) {
        double r = sqrt(x*x + y*y);

        if (r > 10)
            return 0;
        else if (r > 5)
            return 1;
        else if (r > 1)
            return 5;
        else
            return 10;
    }
} // namespace darts

File: solutions/zig/darts/1/darts.zig
const std = @import("std");

pub const Coordinate = struct {
    // This struct, as well as its fields and methods, needs to be implemented.
    x: f32,
    y: f32,
    
    pub fn init(x_coord: f32, y_coord: f32) Coordinate {
        return .{.x = x_coord, .y = y_coord};
    }
    pub fn score(self: Coordinate) usize {
        const r = std.math.sqrt(self.x * self.x + self.y * self.y);
        return if (r <= 1)
            10
            else if (r <= 5)
            5
            else if (r <= 10)
            1
            else
            0;
    }
};
